### PR TITLE
feat: add authentication setting

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

if 'use token authentication' is set to true, oauth2 authentication is disabled

depends on https://github.com/snyk/snyk-ls/pull/544

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: package.json
Comment (teodora-sandu) [inline package.json:114]: Double-checking, but once https://github.com/snyk/snyk-ls/pull/544 is merged then customers updating to the latest VSCode will need to force upgrade their `snyk-ls` (since it updates every 4 days) right?

**Review by teodora-sandu — APPROVED** [review]
As you already know, this doesn't work without https://github.com/snyk/snyk-ls/pull/544. It might be worth waiting until after Tuesday's release to merge this just so we have a bit more time for LS to be updated locally for our customers

File: package.json
Comment (bastiandoetsch) [reply package.json:114]: It will automatically update it, if we increase the protocol version. Probably, we should do that, right?

**Pull request merged**
